Tutorials/GraphOperator/Ising/ising.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import netket as nk
+import networkx as nx
+import numpy as np
+from mpi4py import MPI
+
+sigmax = [[0, 1], [1, 0]]
+sigmaz = [[1, 0], [0, -1]]
+
+mszsz = (np.kron(sigmaz, sigmaz)).tolist()
+
+# Now we define the local operators of our hamiltonian
+# And the sites on which they act
+# Notice that the Transverse-Field Ising model as defined here has sign problem
+L = 20
+site_operator = [sigmax]
+bond_operator = [mszsz]
+
+# Defining a custom graph
+nxg = nx.Graph()
+for i in range(L):
+    nxg.add_edge(i, (i + 1) % L)
+
+# Custom Graph
+g = nk.graph.CustomGraph(nxg.edges())
+
+# Custom Hilbert Space
+hi = nk.hilbert.CustomHilbert(local_states=[1, -1], graph=g)
+
+# Graph Operator
+op = nk.operator.GraphOperator(
+    hi, siteops=site_operator, bondops=bond_operator)
+
+# Restricted Boltzmann Machine
+ma = nk.machine.RbmSpin(hilbert=hi, alpha=1)
+
+# Local Metropolis Sampling
+sa = nk.sampler.MetropolisLocal(machine=ma)
+
+# Gradient Descent
+opt = nk.optimizer.AdaMax()
+
```

Comment:
you are not initializing the machine to random parameters, most likely this is the problem? In `Tutorials/ground_state.py' this is done

```python
# Machine
ma = nk.machine.RbmSpin(hilbert=hi, alpha=1)
ma.init_random_parameters(seed=1234, sigma=0.01)
```

Reply:
Adding it to the `GraphOperator/Ising/ising.py` doesn't fix it. The energies are really low from the first iteration, e.g. -0.5 instead of nearly -20.

Reply:
Interesting... those shouldn't work either :) uhm, more puzzles